src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/network/v2/AwsV2NetworkPluginTest.java

```diff
@@ -0,0 +1,616 @@
+
+		for (String cloudState : cloudStates) {
+			// exercise
+			boolean status = this.plugin.isReady(cloudState);
+
+			// verify
+			Assert.assertFalse(status);
+		}
+	}
+		
+	// test case: Whenever you call the hasFailed method, no matter the value, it
+	// must return false.
+	@Test
+	public void testHasFailed() {
+		// set up
+		String cloudState = ANY_VALUE;
+
+		// exercise
+		boolean status = this.plugin.hasFailed(cloudState);
+
+		// verify
+		Assert.assertFalse(status);
+	}
+	
+	// test case: When calling the getGatewayFromRouteTables method, and a route
+	// list with a gateway ID different of local, it must return a default
+	// destination CIDR block.
+	@Test
+	public void testGetGatewayFromRouteTablesSuccessful() {
+		// set up
+		String gatewayId = ANY_VALUE;
+		String expected = DEFAULT_CIDR_DESTINATION;
+
+		Route route = Route.builder()
+				.gatewayId(gatewayId)
+				.destinationCidrBlock(expected)
+				.build();
+
+		List<Route> routes = new ArrayList<>();
+		routes.add(route);
+
+		// exercise
+		String gateway = this.plugin.getGatewayFromRouteTables(routes);
+
+		// verify
+		Assert.assertEquals(expected, gateway);
+	}
+	
+	// test case: When calling the getGatewayFromRouteTables method, and a route
+	// list with a gateway ID equals to local, it must return a null value.
+	@Test
+	public void testGetGatewayFromRouteTablesUnsuccessful() {
+		// set up
+		String gatewayId = AwsV2NetworkPlugin.LOCAL_GATEWAY_DESTINATION;
+		String expected = null;
+
+		Route route = Route.builder()
+				.gatewayId(gatewayId)
+				.build();
+
+		List<Route> routes = new ArrayList<>();
+		routes.add(route);
+
+		// exercise
+		String gateway = this.plugin.getGatewayFromRouteTables(routes);
+
+		// verify
+		Assert.assertEquals(expected, gateway);
+	}
+	
+	// test case: When calling the getGroupIdBySubnet method, with a valid subnet ID
+	// and client, and a key tag different of groupId, the UnexpectedException will
+	// be thrown.
+	@Test(expected = UnexpectedException.class)
+	public void testGetGroupIdBySubnetUnsuccessful()
+			throws InvalidParameterException, UnexpectedException, InstanceNotFoundException {
+
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		String subnetId = FAKE_SUBNET_ID;
+		String key = ANY_VALUE;
+		String value = ANY_VALUE;
+		Tag tag = buildTags(key, value);
+		Subnet subnet = buildSubnet(subnetId, tag);
+		Mockito.doReturn(subnet).when(this.plugin).getSubnetById(subnetId, client);
+
+		// exercise
+		this.plugin.getGroupIdBySubnet(subnetId, client);
+	}
+
+	// test case: When calling the getSubnetById method, with a valid client, and a
+	// subnet ID invalid, the InstanceNotFoundException will be thrown.
+	@Test(expected = InstanceNotFoundException.class) // verify
+	public void testGetSubnetByIdUnsuccessful()
```

Comment:
shouldn't we have a testGetSubnetByIdSuccessful? We should mock the api returned value to test it

Reply:
I did not find it necessary since the tests of the main methods already cover the success case. And yes, we would have to mock the return value to test.